core/templates/pages/topic-editor-page/navbar/topic-editor-navbar.component.ts

```diff
@@ -132,6 +132,7 @@ export class TopicEditorNavbarComponent {
   discardChanges(): void {
     this.undoRedoService.clearChanges();
     this.discardChangesButtonIsShown = false;
+    this.windowRef.nativeWindow.location.reload();
```

Comment:
Hi @kshitij01042002, I read through the issue thread regarding why reload is used, but just wondering, have you checked if there is a way to achieve the same goal without a hard refresh? Note that hard refresh is quite detremental for the user experience, some users may be on a slow network.

Reply:
@kevintab95 
On your suggestion I tried to find a method that could reset the changes without reloading the page. I noticed that the topic editor page is yet to be migrated, soo working on it was a little difficult for me.
Values in editor are coming for different functions which needed to be called again if we want to reset the values. Reloading the page becomes mandate as similar to exploration editor.
If there's a way we can do it, just drop a hint and I'll try to work it out.

Thanks!

Reply:
"topic editor page is yet to be migrated, soo working on it was a little difficult for me." Could you please explain what exactly you find difficult with the current state of the topic editor?

Also for "Values in editor are coming for different functions which needed to be called again if we want to reset the values.", this seems like a standard thing to do, could you explain why this is tricky?

Reply:
The overall problem is the file is in AngularJS and I understanding those files is a little difficult for me. The migration of this page will be done in this [PR](https://github.com/oppia/oppia/pull/17372). Can I resume my work after this PR is merged?

Soo won't be double work also it'll be easy for me to work on a migrated page as I have experience working on Angular.